official/mnist/mnist.py

```diff
@@ -122,16 +127,49 @@ def model_fn(features, labels, mode, params):
         })
 
 
+def validate_batch_size_for_multi_gpu(batch_size):
+  """For multi-gpu, batch-size must be a multiple of the number of
+  available GPUs.
+
+  Note that this should eventually be handled by replicate_model_fn
+  directly. Multi-GPU support is currently experimental, however,
+  so doing the work here until that feature is in place.
+  """
+  from tensorflow.python.client import device_lib
+
+  local_device_protos = device_lib.list_local_devices()
+  num_gpus = sum([1 for d in local_device_protos if d.device_type == 'GPU'])
```

Comment:
Perhaps handle the case where num_gpus are 0 (with a return) - otherwise we'll see an unhelpful `ZeroDivisionError`?

Reply:
SG-- added below.